uk.ac.kcl.inf.modelling.xdsml.gemoc_henshin/src/uk/ac/kcl/inf/modelling/xdsml/gemoc_henshin/engine/core/HenshinStep.xtend

```diff
@@ -49,10 +50,16 @@ class HenshinStep extends GenericSmallStepImpl {
 			generateMSE(match, match.getRule().getName(), match.toString())
 		} else {
 			var rulesNames = ''
+			var ruleNamesArray = new ArrayList();
 			for (Match m : matches) {
 				var r = m.getRule();
-				rulesNames = rulesNames + ' ' + r.getName()
+				ruleNamesArray.add(r.getName())
 			}
+			var sortedRuleNames = ruleNamesArray.sort()
+			for(String s: sortedRuleNames){
+				rulesNames = rulesNames + ' ' + s
+			}				
+			
```

Comment:
A more compact way of saying the same in Xtend would be using lambdas, roughly like this:

```
val rulesNames = matches.map[m | m.rule.name].sort.join(' ')
```